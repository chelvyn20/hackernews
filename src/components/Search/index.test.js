import { createRoot } from 'react-dom/client';
import { create, act } from 'react-test-renderer';
import { Search } from './index';

describe('Search', () => {
  const props = {
    value: 'react',
    onChange: jest.fn(),
    onSubmit: jest.fn(),
  };

  let container;

  beforeEach(() => {
    container = document.createElement('div');
    document.body.appendChild(container);
  });

  afterEach(() => {
    document.body.removeChild(container);
    container = null;
  });

  it('should render, show input value as "react" and button textContent as Search', () => {
    act(() => {
      createRoot(container).render(<Search {...props}>Search</Search>);
    });
    const input = container.querySelector('input');
    expect(input.value).toBe('react');

    const button = container.querySelector('button');
    expect(button.textContent).toBe('Search');
  });

  test('snapshots', () => {
    const component = create(<Search {...props}>Search</Search>);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
