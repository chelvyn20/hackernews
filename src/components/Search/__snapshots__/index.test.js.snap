// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search snapshots 1`] = `
<form
  onSubmit={[MockFunction]}
>
  <input
    onChange={[MockFunction]}
    type="text"
    value="react"
  />
  <button
    type="submit"
  >
    Search
  </button>
</form>
`;
